src/rest/handlers/rest_edges.erl

```diff
@@ -118,3 +107,65 @@ parse_bulk_request(#{<<"sourceZone">> := SourceZone, <<"destinationZone">> := De
     end;
 parse_bulk_request(_) ->
     {error, "Invalid JSON"}.
+
+
+%%%---------------------------
+%% operation executor
+%%%---------------------------
+
+-spec execute_operation(Op :: atom(), From :: binary(), To :: binary(), Permissions :: binary() | undefined,
+    Trace :: binary(), Successive :: boolean()) -> ok | {error, any()}.
+execute_operation(Op, From, To, Permissions, Trace, false) ->
+    try
+        {ok, {FromZone, _}} = gmm_utils:split_vertex_id(From),
+        ZoneId = list_to_binary(?ZONE_ID),
+        if
+            %% @todo ZONE_ID is list "zone0", and not a binary <<"zone0">> - I need to do something about it
+            FromZone =/= ZoneId -> throw({return,
```

Comment:
rozumiem ideę, ale załatwianie tego przez rzucanie wyjątku średnio mi się podoba, choć jeśli nam się śpieszy, jestem skłonny dać approve, najwyżej później się spróbuje załatwić to ładniej

Reply:
Rozumiem. Ogólnie to ten wyjątek ma imitować operator "return", plus skrócić sprawdzania warunków w różnych miejscach. Tzn. bez tego musiałbym tu zrobić sporą refaktoryzację, w szczególności mam problem z tym, że tu by było z 10 zagnieżdzonych "case of" albo z 10 małych funkcji, postaci

`
fun_1(Arg1, ...) ->
  case Cond1 of
    true -> fun_2(Arg1, ...);
    _ -> {error, something}
  end.

fun_2(Arg1, ...) ->
  case Cond2 of
    true -> fun_3(Arg1, ...);
    _ -> {error, something}
  end.

...
`

Reply:
dobra,to zostawmy to chwilowo, skoro czas nas nagli